remote/remote_state_s3.go

```diff
@@ -674,20 +676,23 @@ func EnableSSEForS3BucketWide(s3Client *s3.S3, config *RemoteStateConfigS3, terr
 }
 
 // Enable bucket-wide Access Logging for the AWS S3 bucket specified in the given config
-func EnableAccessLoggingForS3BucketWide(s3Client *s3.S3, config *RemoteStateConfigS3, terragruntOptions *options.TerragruntOptions, logsBucket string) error {
+func EnableAccessLoggingForS3BucketWide(s3Client *s3.S3, config *RemoteStateConfigS3, terragruntOptions *options.TerragruntOptions, logsBucket string, logsBucketPrefix string) error {
 	if err := configureBucketAccessLoggingAcl(s3Client, aws.String(logsBucket), terragruntOptions); err != nil {
 		return errors.WithStackTrace(err)
 	}
 
-	targetPrefix := "TFStateLogs/"
-	terragruntOptions.Logger.Debugf("Putting bucket logging on S3 bucket %s with TargetBucket %s and TargetPrefix %s", config.Bucket, logsBucket, targetPrefix)
+	if logsBucketPrefix == "" {
+		logsBucketPrefix = "TFStatelogs/"
+	}
```

Comment:
OK, a few sanity checks here:

1. Does implementing it this way mean you can't set the value to an empty string? If so, we may need to set this default at parse time, when we can tell the difference between "user set the value explicitly to an empty string" or "user didn't set this value at all."
1. NIT: Don't you have a constant declared for this value already?
1. NIT: extra whitespace before the closing curly brace?

Reply:
Hey Jim - you're absolutely right! I went and chose the most straightforward way to implement it, but all of the points you've made are valid. 
1. I actually misunderstood your first comment, in this case, apologies! As before, we weren't giving the user any choice, I assumed that's what you meant by achieving backwards compatibility! _So, totally agree - let's put it at parse time of the config - I think the right place is `parseExtendedS3Config`_
2. Wasn't one just yet - but it's worth it at this time!
3. Hm, my IDE doesn't show it and `go fmt` ran fine. I'll make sure I double-check again before pushing!